util/stream.go

```diff
@@ -63,10 +85,10 @@ func NewMessageStream(conn net.Conn, parser Parser) *MessageStream {
 
 	go m.outbound()
 	go m.inbound()
-
-	for i := 0; i < 25; i++ {
+	for i := 0; i < numParserGoroutines; i++ {
```

Comment:
The parser goroutines are started here. I've added a constant to make sure we clean up as many goroutines as we create.